extensions/sql-migration/src/models/stateMachine.ts

```diff
@@ -196,15 +201,61 @@ export class MigrationStateModel implements Model, vscode.Disposable {
 				}
 			}
 		});
-
 		this._assessmentResults = {
 			issues: serverLevelAssessments,
 			databaseAssessments: databaseLevelAssessments
 		};
-
+		this.generateAssessmentTelemetry(startTime, endTime, connectionProfile);
 		return this._assessmentResults;
 	}
 
+	private async generateAssessmentTelemetry(startTime: number, endTime: number, connectionProfile: azdata.connection.ConnectionProfile): Promise<void> {
+		try {
+			const queryProvider = azdata.dataprotocol.getProvider<azdata.QueryProvider>(connectionProfile.providerId, azdata.DataProviderType.QueryProvider);
+			const query = 'SELECT name, compatibility_level FROM sys.databases; ';
+			const results = await queryProvider.runQueryAndReturn(await (azdata.connection.getUriForConnection(this.sourceConnectionId)), query);
+			const compatLevelHashMap: Map<string, string> = new Map();
+			results.rows.forEach(r => {
+				compatLevelHashMap.set(r[0].displayValue, r[1].displayValue);
+			});
+
+			sendSqlMigrationActionEvent(
+				TelemetryViews.MigrationWizardTargetSelectionPage,
+				TelemetryAction.ServerAssessment,
+				{
+					'sessionId': this._sessionId,
+					'tenantId': this._azureAccount.properties.tenants[0].id,
+					'hashedServerName': hashString((connectionProfile.serverName))
+				},
+				{
+					'issuesCount': this._assessmentResults.issues.length,
+					'warningsCount': this._assessmentResults.databaseAssessments.reduce((count, d) => count + d.issues.length, 0),
+					'duration': endTime - startTime,
+					'databaseCount': this._assessmentResults.databaseAssessments.length,
```

Comment:
is this metric the number of databases on the server or number of assessed databases... could it be different if the assessment of a database fails?